plugins/mail/nullmailer_queue

```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
+
+: <<=cut
+
+=head1 NAME
+
+nullmailer_queue - monitor for nullmailer's send queue.
+
+=head1 APPLICABLE SYSTEMS
+
+This plugin is applicable to any system running the nullmailer MTA.
+
+=head1 CONFIGURATION
+
+This plugin needs to be able to read the nullmailer queue, so it
+will need to run as that user or root.
+
+Additionally, the nullmailer queue directory may vary; in that case you
+can specify it with the "queuedir" and "errordir" environment variables.
+
+Example:
+
+[nullmailer_queue]
+user nullmail
+
+env.queuedir /var/spool/nullmailer/queue
+env.errordir /var/spool/nullmailer/failed
+
+=head1 INTERPRETATION
+
+This plugin will draw a stack of 2: the number of emails in the queue, and
+the number of emails that have permanently failed sending. The latter has
+a warning attached, since those should never happen.
```

Comment:
Just for my understanding:
* number of emails in queue: this is a *state*
* number of permanently rejected emails: this feels like a *rate* to me (emails per seconds)
    * does this value grow forever? (with manual cleanup)

Does this combination result in an understandable graph?
Maybe two separate graphs would be better? (maybe produced by a single "multigraph" plugin)

Reply:
Number of permanently rejected emails is a _state_, and is ever increasing with manual cleanup as you said. Think of it as the `hold` value for the upstream `postfix` plugin.

I chose this version, since there is, under normal operation, no reason that this should ever be non-zero, and anything else requires administrator intervention, since it is most likely a configuration error.